dChatFilter/dChatFilter.cpp

```diff
@@ -105,9 +105,22 @@ bool dChatFilter::IsSentenceOkay(const std::string& message, int gmLevel) {
 	std::string segment;
 	std::regex reg("(!*|\\?*|\\;*|\\.*|\\,*)");
 
+
+		
+
+	#ifdef _win32 
+	while (std::getline(file, line)) {
+		line.erase(std::remove(line.begin(), line.end(), '\r'), line.end()); //Remove nix line-endings
+		std::transform(line.begin(), line.end(), line.begin(), ::tolower); //Transform to lowercase
+		m_Words.push_back(CalculateHash(line));
+	}
+	#else 
 	while (std::getline(sMessage, segment, ' ')) {
+	#endif
+
+
 		std::transform(segment.begin(), segment.end(), segment.begin(), ::tolower); //Transform to lowercase
-		segment = std::regex_replace(segment, reg, "");
+			segment = std::regex_replace(segment, reg, "");
```

Comment:
The erase should be unconditional of platform. This will allow use of `\r\n` line-endings on Unix

Reply:
New to this pull request on github. Am i the only one that can make commits?